question20/main.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+const wrapper = document.getElementById('js-wrapper');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'loading';
+  wrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => {
+  document.getElementById('loading').remove();
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/1f12787c-e5e9-4cfa-9c1d-2331884ca8ec'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTabaleElement(userData);
+    } else {
+    wrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+const userDataItems = {
+  userId: "ID",
+  name: "名前",
+  gender: "性別",
+  age: "年齢"
+}
+
+const renderTabaleElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  const userDataKeys = Object.keys(userDataItems);
+  const tableHeadValue = Object.values(userDataItems);
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableDataElement(userData,userDataKeys));
```

Comment:
IMO
For this table, I think it would be more appropriate to put the tr in thead instead of in tbody. Because it is a heading.